ical_exporter.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import "io" 
+
+type ICalTaskExporter struct {
+	Version string 
+	PRODID string 
+	CALSCALE string
+	TimeFormat string // UTC time format string (layout string passed to time.Format)
+}
+
+// tryWriteStrings is a helper function used to write strings to a writer if an error has not previously occured
+// 	if an error occurs during the write the "error" argument will receive the error 
+func tryWriteStrings(error *error, writer io.Writer, strings ...string) {
+	if strings != nil {
+		for _, str := range strings {
+			if *error != nil {
+				return
+			}
+			_, *error = io.WriteString(writer, str)
+		}
+	}
+}
+
+func (iCalTaskExporter ICalTaskExporter) getVersion() string {
+	if iCalTaskExporter.Version == "" {
+		return "2.0"
+	}
+	return iCalTaskExporter.Version
+}
+
+func (iCalTaskExporter ICalTaskExporter) getPRODID() string {
+	if iCalTaskExporter.PRODID == "" {
```

Comment:
what's the PRODID? i'm missing context from ical

Reply:
PRODID: This property specifies the identifier for the product that created the iCalendar object.
Needs to be unique and seems to be mandatory. 

https://www.kanzaki.com/docs/ical/prodid.html